core/payments/reservation/reservation_vault_monitor.go

```diff
@@ -101,6 +97,54 @@ func (vm *ReservationVaultMonitor) refreshReservations(ctx context.Context) erro
 	return nil
 }
 
+// Fetches reservations from the PaymentVault. If number of accountIDs exceeds configured rpcBatchSize, multiple RPC
+// calls will be made in series to fetch all reservation data. If rpcBatchSize is configured to be 0, all data
+// will be fetched in a single call, no matter how many accounts are passed in.
+func (vm *ReservationVaultMonitor) fetchReservations(
+	ctx context.Context,
+	accountIDs []gethcommon.Address,
+) ([]*bindings.IPaymentVaultReservation, error) {
+	if len(accountIDs) == 0 {
+		return nil, nil
+	}
+
+	// Split accounts into accountBatches to avoid RPC size limits
+	var accountBatches [][]gethcommon.Address
+
+	// Special case: 0 means no batching
+	if vm.rpcBatchSize == 0 {
+		accountBatches = [][]gethcommon.Address{accountIDs}
+	} else {
+		// Create batches of the specified size
+		for i := 0; i < len(accountIDs); i += int(vm.rpcBatchSize) {
+			end := i + int(vm.rpcBatchSize)
+			if end > len(accountIDs) {
+				end = len(accountIDs)
+			}
+			accountBatches = append(accountBatches, accountIDs[i:end])
+		}
+	}
+
+	allReservations := make([]*bindings.IPaymentVaultReservation, 0, len(accountIDs))
+	for batchIndex, batch := range accountBatches {
+		newReservations, err := vm.paymentVault.GetReservations(ctx, batch)
```

Comment:
nit: this would be a good usecase for errgroup no? Fetch everything in parallel, if anything errors error the whole thing since we need all the accounts anyways.

Reply:
bc278dd92fde8f5c57bcd28032363cf64b80cfc8

👍 Done. Feels a bit overengineered, but not too bad to reason about, and makes this fully future proof